tests/unit/s2n_server_new_session_ticket_test.c

```diff
@@ -72,6 +72,22 @@ static int s2n_setup_test_keys(struct s2n_connection *conn, struct s2n_config *c
     return S2N_SUCCESS;
 }
 
+bool s2n_test_session_ticket_callback_flag = false;
+static int s2n_test_session_ticket_callback(struct s2n_connection *conn,
+                                         uint8_t *session_id_data, size_t session_id_len,
+                                         uint8_t *session_data, size_t session_len,
+                                         uint32_t session_lifetime)
+{
+    EXPECT_EQUAL(session_id_len, 0);
+    EXPECT_NULL(session_id_data);
+    EXPECT_TRUE(session_len > 0);
+    EXPECT_NOT_NULL(session_data);
+    EXPECT_TRUE(session_lifetime > 0);
```

Comment:
Would it be worthwhile to also assert that all these values match the values of calling the APIs here?

Reply:
Or maybe more correctly, we should assert that the values in the callback match the values after a full handshake.